### PR TITLE
Sparse storage

### PR DESCRIPTION
**Pull request by FrancescAlted**
New `blosc2_storage` struct.  This opens the door to the implementation of a sparse persistent storage as per our ROADMAP.

### CODE REVIEW

File: blosc/blosc2.h [hunk 1]
```diff
@@ -900,6 +900,21 @@ BLOSC_EXPORT int blosc2_getitem_ctx(blosc2_context* context, const void* src,
...
 #define BLOSC2_METALAYER_NAME_MAXLEN 31
 
+/**
+ * @brief This struct is meant for holding storage parameters for a
+ * for a blosc2 container, allowing to specify, for example, how to interpret
+ * the contents included in the schunk.
+ */
+typedef struct {
+    bool sparse;  //!< Whether the chunks are sparse or sequential (frame)
+    char* path;   //!< The path for storage; if NULL, that means in-memory
```
Comment (aleix11alcacer) [inline blosc/blosc2.h:910]: Add a 'else' description?

File: blosc/schunk.c [hunk 2]
```diff
@@ -95,6 +95,11 @@ blosc2_schunk *blosc2_new_schunk(blosc2_cparams cparams, blosc2_dparams dparams,
   dparams.schunk = schunk;
   schunk->dctx = blosc2_create_dctx(dparams);
 
+  // Storage info.  We use defaults here because we don't want to break
+  // the existing `blosc2_new_schunk()` API.
+  schunk->storage = (blosc2_storage*)malloc(sizeof(blosc2_storage));
+  memcpy(schunk->storage, &BLOSC2_STORAGE_DEFAULTS, sizeof(blosc2_storage));
+
   schunk->frame = frame;
   if (frame != NULL) {
```
Comment (aleix11alcacer) [inline blosc/schunk.c:104]: You should update `schunk->storage` params to indicate that a frame (persistent or not) is being used.

File: blosc/schunk.c [hunk 3]
```diff
@@ -112,14 +117,20 @@ blosc2_schunk *blosc2_new_schunk(blosc2_cparams cparams, blosc2_dparams dparams,
...
-    return NULL;
+                                   blosc2_storage* storage) {
+  if (storage == NULL) {
+    storage = (blosc2_storage*)malloc(sizeof(blosc2_storage));
+    memcpy(storage, &BLOSC2_STORAGE_DEFAULTS, sizeof(blosc2_storage));
   }
 
   blosc2_schunk* schunk = calloc(1, sizeof(blosc2_schunk));
 
+  if (storage->sparse == false) {
```
Comment (aleix11alcacer) [inline blosc/schunk.c:128]: Create an empty persistent schunk is not allowed yet.

**Review by aleix11alcacer — CHANGES_REQUESTED** [review]

File: blosc/blosc2.h (hunk 1 above)
Comment (FrancescAlted) [reply blosc/blosc2.h]: I don't think is necessary, as the 'else' is described at the beginning.  But I added a bit more detailed description in 8db4718.

File: blosc/schunk.c (hunk 2 above)
Comment (FrancescAlted) [reply blosc/schunk.c:104]: Addressed in 8db4718

File: blosc/schunk.c (hunk 3 above)
Comment (FrancescAlted) [reply blosc/schunk.c:132]: Addressed in 8db4718.

**Comment by FrancescAlted** [comment]
This should be good for merging now.

**Pull request merged**